tests/validation_tests/test_failures.py

```diff
@@ -1,17 +1,19 @@
+import pandas as pd
 import pytest
 
 from core.errors import UnimplementedUCException
 from core.validation.failures import (
     InvalidEnumValueFailure,
     NoInputFailure,
     NonNullableConstraintFailure,
+    NonUniqueCompositeKeyFailure,
     WrongInputFailure,
     WrongTypeFailure,
     serialise_user_centered_failures,
 )
 
 
-def test_invalid_enum_user_centered_failures():
+def test_serialise_user_centered_failures():
```

Comment:
Think we should have 2 tests:
- standard validation failures output is as expected - this should cover different failures from various tabs to ensure the nesting logic is working as expected
- pre-transformation failures output is as expected as this flow is slightly different